### PR TITLE
[TOS-983] fix(action-step-form) : adding if condition for setting testDataProfileStepId

### PR DESCRIPTION
**Pull request by nixonakash01**
# Description
* Jira : https://testsigma.atlassian.net/browse/TOS-983
Even if TDP is mapped, while running the test its shows TDP is not available

The test data profile is mapped to the test case yet while running the test it shows NO TDP is associated.
This happens with a particular test case but if the same TDP is mapped to other test it works
# Fix
added a if condition so that testDataProfileStepId is serialised only if the testStep is of type 'FOR_LOOP'

### CODE REVIEW

File: ui/src/app/components/webcomponents/action-step-form.component.ts [hunk 1]
```diff
@@ -1921,7 +1921,8 @@ export class ActionStepFormComponent extends BaseComponent implements OnInit {
         this.assignDataValue(this.getDataTypeString(TestDataType.parameter, data));
       }
       else if( typeof(data)==="object" ){
-        this.testStep.testDataProfileStepId = data?.testDataProfileStepId;
+        if(this.testStep.type==="FOR_LOOP")
```
Comment (guganabu) [inline ui/src/app/components/webcomponents/action-step-form.component.ts:1924]: plz remove hard coded matching @nixonakash01

Comment (nixonakash01) [reply ui/src/app/components/webcomponents/action-step-form.component.ts]: done bro check latest commit

File: ui/src/app/components/webcomponents/action-step-form.component.ts [hunk 2]
```diff
@@ -1921,7 +1921,8 @@ export class ActionStepFormComponent extends BaseComponent implements OnInit {
         this.assignDataValue(this.getDataTypeString(TestDataType.parameter, data));
       }
       else if( typeof(data)==="object" ){
-        this.testStep.testDataProfileStepId = data?.testDataProfileStepId;
+        if(this.testStep.type===TestStepType.FOR_LOOP)
+          this.testStep.testDataProfileStepId = data?.testDataProfileStepId;
```
Comment (tarun-testsigma) [inline ui/src/app/components/webcomponents/action-step-form.component.ts:1925]: @nixonakash01 Ragul has some changes related to FOR LOOP and I would suggest you to re verify this after his PR is merged

Comment (nixonakash01) [reply ui/src/app/components/webcomponents/action-step-form.component.ts:1925]: okay bro

**Pull request merged**